needls/server.py

```diff
@@ -334,9 +334,16 @@ def completions(ls, params: CompletionParams = None):
     if not ls.needs_store.needs_initialized:
         return []
 
-    lines, word = get_lines_and_word(ls, params)
-    line_number = params.position[0]
-    line = lines[line_number]
+    try:
+        lines, word = get_lines_and_word(ls, params)
+        line_number = params.position[0]
+        line = lines[line_number]
+    except IndexError:
+        # vs code has default keyboard shortcuts for trigger suggest: ctrl + space,
+        # if user use this shortcut will trigger completion, if used at a empty line,
+        # then the function above to get lines and words will throw out of index error,
+        # hence, this exception here is needed.
```

Comment:
Can't we check if a line is empty before we execute get_lines_and_word?

Catching exceptions is more for catching an unexpected problem.
But with an "empty line" we know the reason and should check for it.